### ISSUE TITLE
Accepted browser language cannot be correctly detected

### ISSUE THREAD

**Issue by arnisjuraga** [opencart/opencart#5301]
Modern browsers uses additional priority weight value in a header.
In a result, `startup.php` will not detect browser language correctly:

e.g. 
https://github.com/opencart/opencart/blob/master/upload/catalog/controller/startup/startup.php#L57

will return:

~~~
array(4) { 
    [0]=> string(5) "en-US" 
    [1]=> string(8) "en;q=0.8" 
    [2]=> string(8) "lv;q=0.6" 
    [3]=> string(8) "lt;q=0.4"
}
~~~

in a result, `in_array` will not return true for any strings with `q` parameter.

http://stackoverflow.com/questions/6157485/content-language-and-accept-language
https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.4

**Comment by danielkerr** [comment]
look m* u might think you know everything but when it comes to open source and having the language system working on as many system as possible it does work so well!

the current system is fine and has been working fine for many years!

now stop trying to post changes that will cause issues for many users!

**Comment by danielkerr** [comment]
i dont why u even posted this crap as an issue!

**Comment by arnisjuraga** [comment]
I have done tests, the I have already posted. Just check out these:

https://forum.opencart.com/viewtopic.php?f=161&t=113954
https://forum.opencart.com/viewtopic.php?f=191&t=143372
https://forum.opencart.com/viewtopic.php?f=20&t=52319

**Comment by arnisjuraga** [comment]
it IS and WILL work only for the first language in `$browser_languages` array, because it does not have quality score added. You can't match "en-US" with "en-us;q=0.5", right? 

if you want to serve ond use one and only language for the site, o ruse en as the only "primary" language, as most of the Opencart users - then it's not a problem and the site will work well. Just read my code, read forum, and read wiki.

As I told - you will not have a problems, if your site "default" language is set to 'en', because most of the users has English as primary acceptable browser language.

**Comment by iplocker** [comment]
Hello.
**I have the issue** as well , try a default 2.3.0.2 installation and install Greeks, always when I visit the site at front I am taking English language even if I have Greeks as default language .
Thanks arnisjuraga your fix with the startup.php **it works**.
Thanks

**Comment by zemguru** [comment]
https://github.com/opencart/opencart/issues/5301#issuecomment-289044132

**Comment by iplocker** [comment]
@ danielkerr  this fix is for 2.3.02 version of OC ?
